File: DeletesCellsDelegate/TableSample/ViewController.m
//
//  ViewController.m
//  SecondVCDeletesCells
//
//  Created by Javier Calderon Jr. on 12/29/19.
//  Copyright © 2019 RockefellerFiles. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


-(void)viewWillDisappear:(BOOL)animated
{
    [super viewWillDisappear:animated];
    if (self.isMovingFromParentViewController) {
        [self.delegate deleteRowInTableview:1];
    }
}


-(IBAction)moveToListview:(id)sender
{
    [self.delegate deleteRowInTableview:1];
}
@end

File: SecondVCDeletesCells/SecondVCDeletesCells/ViewController.swift
//
//  ViewController.swift
//  SecondVCDeletesCells
//
//  Created by Javier Calderon Jr. on 12/29/19.
//  Copyright © 2019 RockefellerFiles. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var delegate: DeleteRowInTableviewDelegate?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        if isMovingFromParent {
            delegate!.deleteRow(inTableview: 1)
        }
    }
    
    @IBAction func move(toListview sender: Any) {
        delegate!.deleteRow(inTableview: 1)
    }
}
